File: frontend/src/types/message.ts
export type MessageType = 
  | 'join_game'
  | 'start_game'
  | 'select_card'
  | 'game_state'
  | 'card_revealed'
  | 'round_end'
  | 'game_end'
  | 'error';

export interface Message {
  type: MessageType;
  payload: unknown;
}

export interface SelectCardPayload {
  cardIndex: number;
  useChopsticks: boolean;
  secondCardIndex?: number;
}

File: backend/models/message.go
package models

import "encoding/json"

// MessageType represents the type of WebSocket message
type MessageType string

const (
	MsgTypeJoinGame     MessageType = "join_game"
	MsgTypeStartGame    MessageType = "start_game"
	MsgTypeSelectCard   MessageType = "select_card"
	MsgTypeGameState    MessageType = "game_state"
	MsgTypeCardRevealed MessageType = "card_revealed"
	MsgTypeRoundEnd     MessageType = "round_end"
	MsgTypeGameEnd      MessageType = "game_end"
	MsgTypeError        MessageType = "error"
)

// Message represents a WebSocket message
type Message struct {
	Type    MessageType     `json:"type"`
	Payload json.RawMessage `json:"payload"`
}

// SelectCardPayload represents the payload for card selection
type SelectCardPayload struct {
	CardIndex       int  `json:"cardIndex"`
	UseChopsticks   bool `json:"useChopsticks"`
	SecondCardIndex *int `json:"secondCardIndex,omitempty"`
}
